### PR TITLE
Dyn-4762 get started onboard guide

### PR DESCRIPTION
**Pull request by filipeotero**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

This PR implements the first step of the Onboarding Guide.
Some features that were also added in this PR:
1. A link at the survey steps redirects to the onboarding guide.
2. WebBrowser rendering an empty image that can be replaced by a gif in the future.

OnboardingGuideFirstStep

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

@QilongTang 

### FYIs

@RobertGlobant20

### CODE REVIEW

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs [hunk 1]
```diff
@@ -415,7 +415,8 @@ private Step CreateStep(Step jsonStepInfo, HostControlInfo hostControlInfo, int
                         {
                             FormattedText = formattedText,
                             Title = title
-                        }
+                        },
+                        GuidesManager = this
```
Comment (filipeotero) [inline src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs:419]: This was set to launch the Onboarding Guide once the link of the survey's step is clicked

File: src/DynamoCoreWpf/UI/GuidedTour/Survey.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ public class Survey : Step
 
         public bool IsRatingVisible { get; set; }
 
+        internal GuidesManager GuidesManager { get; set; }
```
Comment (filipeotero) [inline src/DynamoCoreWpf/UI/GuidedTour/Survey.cs:16]: Same here

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 3]
```diff
@@ -61,8 +61,9 @@ public partial class DynamoView : Window, IDisposable
         public const string BackgroundPreviewName = "BackgroundPreview";
 
         //The "Packages" and "Get Started" strings needs to be hardcoded due that are hardcoded in the json file (no need localization)
-        private static string GetStartedGuideName = "Get Started";
-        private static string PackagesGuideName = "Packages";
+        internal static string GetStartedGuideName = "Get Started";
+        internal static string PackagesGuideName = "Packages";
+        internal static string OnboardingGuideName = "Packages";
```
Comment (filipeotero) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:67]: These access modifiers were changed to be accessed in the SurveyPopupWindow class when the link is clicked.

File: src/DynamoCoreWpf/Views/GuidedTour/SurveyPopupWindow.xaml.cs [hunk 4]
```diff
@@ -22,15 +22,18 @@ public SurveyPopupWindow(SurveyPopupViewModel viewModel, HostControlInfo hInfo)
...
+            
             //Setting the host over which the popup will appear and the placement mode
             PlacementTarget = hInfo.HostUIElement;
             Placement = hInfo.PopupPlacement;
 
             //When setting a value for offset it will move the popup taking as a initial position the host control position in which the popup is shown
             HorizontalOffset = hInfo.HorizontalPopupOffSet;
             VerticalOffset = hInfo.VerticalPopupOffSet;
+
+            Focus();
```
Comment (filipeotero) [inline src/DynamoCoreWpf/Views/GuidedTour/SurveyPopupWindow.xaml.cs:36]: This focus was added to identify the hand cursor above the link

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 5]
```diff
@@ -61,8 +61,9 @@ public partial class DynamoView : Window, IDisposable
...
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:66]: The string name does not seem correct here? `Packages` or `Onboarding`?

**Review by QilongTang — COMMENTED** [review]
Some comment, then LGTM

File: src/DynamoCoreWpf/Views/GuidedTour/SurveyPopupWindow.xaml [hunk 6]
```diff
@@ -119,15 +120,37 @@
...
-                <!--Link to Next Guide-->
+
+                <!-- This grid represents the link that redirects to the onboarding guide -->
+                <Grid Grid.Row="5" Name="GetStartedLink">
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="40"></ColumnDefinition>
+                        <ColumnDefinition Width="*"></ColumnDefinition>
+                        <ColumnDefinition Width="*"></ColumnDefinition>
+                        <ColumnDefinition Width="10"></ColumnDefinition>
+                    </Grid.ColumnDefinitions>
```
[the remark this answers is not in the document]
Comment (filipeotero) [reply src/DynamoCoreWpf/Views/GuidedTour/SurveyPopupWindow.xaml:131]: Here is an example of the text converted into Russian:
image

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs (hunk 5 above)
Comment (filipeotero) [reply src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs]: That's incorrect. Fixed now. Thanks!

**Comment by QilongTang** [comment]
I think this looks solid for me. Glad we can reuse the frame work. Did we have any unit tests covering guided tour?

**Comment by filipeotero** [comment]
> I think this looks solid for me. Glad we can reuse the frame work. Did we have any unit tests covering guided tour?

Yes! It was made by @RobertGlobant20 🙂

**Comment by QilongTang** [comment]
> > I think this looks solid for me. Glad we can reuse the frame work. Did we have any unit tests covering guided tour?
> 
> Yes! It was made by @RobertGlobant20 🙂

 Do you think we can add any unit tests covering the newly added tour?

**Comment by filipeotero** [comment]
> > > I think this looks solid for me. Glad we can reuse the frame work. Did we have any unit tests covering guided tour?
> > 
> > 
> > Yes! It was made by @RobertGlobant20 🙂
> 
> Do you think we can add any unit tests covering the newly added tour?

Sure, we can check if the popup is opened and if the text is according to the JSON file. What do you think?

**Comment by QilongTang** [comment]
> > > > I think this looks solid for me. Glad we can reuse the frame work. Did we have any unit tests covering guided tour?
> > > 
> > > 
> > > Yes! It was made by @RobertGlobant20 🙂
> > 
> > 
> > Do you think we can add any unit tests covering the newly added tour?
> 
> Sure, we can check if the popup is opened and if the text is according to the JSON file. What do you think?

Thanks, please do!

**Comment by filipeotero** [comment]
Adding a new feature that creates a new workspace when starting the onboarding guide from the menu (Saving the current workspace if any).
CreateNewWorkspaceOnboarding

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@reddyashish @zeusongit Any comments? If not, I think this is a good starting point for the new tour

**Review by zeusongit — APPROVED** [review]
LGTM as well

**Pull request merged**